### PR TITLE
increase hit areas for toolbar buttons

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/392891325557410/1198872510442766
Tech Design URL:
CC:

**Description**:

Use a toolbar subclass to handle hit tests.

Increasing the actual size of the buttons causes the buttons to not be laid out correctly due to the custom views.

Not worrying about iPad layout at this stage.

**Steps to test this PR**:

1. Launch the app on a phone (or iPad with phone layout), ensure tab switcher and bookmarks button work as planned and are not triggered when they shouldn't be (e.g. browsing, scrolling, etc).

**OS Testing**:

* [x] iOS 11
* [x] iOS 12
* [x] iOS 13
* [x] iOS 14

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/GestureToolbarButton.swift [hunk 1]
```diff
@@ -128,14 +128,7 @@ class GestureToolbarButton: UIView {
...
 
     override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
-        guard let touch = touches.first else {
-            imposeReleaseAnimation()
-            return
-        }
-        guard point(inside: touch.location(in: self), with: event) else {
-            imposeReleaseAnimation()
-            return
-        }
```
Comment (brindy) [inline DuckDuckGo/GestureToolbarButton.swift:138]: All the above is largely unnecessary - the main scenario it was for is if the user presses down, then moves, then releases, but by that point the long press gesture has already kicked in.

File: DuckDuckGo/TabSwitcherButton.swift [hunk 2]
```diff
@@ -155,9 +155,6 @@ class TabSwitcherButton: UIView {
 
         workItem?.cancel()
         guard workItem != nil else { return }
-
-        guard let touch = touches.first else { return }
-        guard point(inside: touch.location(in: self), with: event) else { return }
```
Comment (brindy) [inline DuckDuckGo/TabSwitcherButton.swift:160]: Same as above.

**Review by samsymons — APPROVED** [review]
Awesome, this works great. Tested typical usage as well as with VoiceOver on, just to be safe. 👏

**Comment by brindy** [comment]
Thanks @samsymons !

**Pull request merged**